internal/test/random.go

```diff
@@ -0,0 +1,60 @@
+package test
+
+import (
+	"crypto/rand"
```

Comment:
I suggest we use [math/rand](https://pkg.go.dev/math/rand) package instead. It would allow for [seeding](https://pkg.go.dev/math/rand#Rand.Seed) tests and making them reproducible.

Reply:
Initially I implemented the `Random` funcs with `math/rand` and then ran in to golangci-lint errors. That's when I pivoted to `crypto/rand`. However, I made all the changes as instructed. Will push the code when I get off work here shortly. 

Thank you for the review.